### PR TITLE
Logging: Report Post Windows 10 2004 versions, like Windows 11

### PR DESCRIPTION
**Pull request by Docteh**
Qt5 and Qt6 don't really do a good job of reporting Windows versions past the 2004 version.

Current: Windows 10 Version 2009
This Patch: Windows 10 Version 21H1 (Build 19043.1706)
Also: Windows 11 Version 21H2 (Build 22000.675)

Fixes: #8362

------
Just `Windows 11 Version 21H2` might be sufficient, but its easier to quickly delete some code than quickly add some code.

### CODE REVIEW

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -198,6 +198,31 @@ static void RemoveCachedContents() {
...
+static QString PrettyProductName() {
+#ifdef _WIN32
+    // After Windows 10 Version 2004, Microsoft decided to switch to a different notation: 20H2
+    // With that notation change they changed the registry key used to denote the current version
+    QSettings windows_registry(
+        QStringLiteral("HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion"),
+        QSettings::NativeFormat);
+    const QString release_id = windows_registry.value(QStringLiteral("ReleaseId")).toString();
+    if (release_id == QStringLiteral("2009")) {
+        const uint current_build = windows_registry.value(QStringLiteral("CurrentBuild")).toUInt();
```
Comment (lioncash) [inline src/yuzu/main.cpp:210]:
```suggestion
        const u32 current_build = windows_registry.value(QStringLiteral("CurrentBuild")).toUInt();
```

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -198,6 +198,31 @@ static void RemoveCachedContents() {
...
+        const QString display_version =
+            windows_registry.value(QStringLiteral("DisplayVersion")).toString();
+        const uint ubr = windows_registry.value(QStringLiteral("UBR")).toUInt();
```
Comment (lioncash) [inline src/yuzu/main.cpp:213]:
```suggestion
        const u32 ubr = windows_registry.value(QStringLiteral("UBR")).toUInt();
```

File: src/yuzu/main.cpp [hunk 3]
```diff
@@ -198,6 +198,31 @@ static void RemoveCachedContents() {
...
+        uint version = 10;
```
Comment (lioncash) [inline src/yuzu/main.cpp:214]:
```suggestion
        u32 version = 10;
```

Please don't rely on `uint` all lowercase as a typedef,

Comment (Docteh) [reply src/yuzu/main.cpp:214]: sorry, slow learner sometimes

**Comment by Docteh** [comment]
Just some notes

Someone posted a log a few days ago that showed "Windows 8 Version 6.2 (Build 22621)" 

It's actually a Windows 11 build, but the user recently updated from Windows 8 to 11, but looking at it, I think I should delete "OS"

I looked at the current code in qtbase, and it looks like in what will become qt 6.4.0 will stop just reporting "Windows 1x Version 2009"

**Comment by Docteh** [comment]
that last force push was deleting "OS" from the log line, and also a rebase. Just noting it in case I hit the "Compare" button a third time.

**Comment by Docteh** [comment]
Update, The 20220602 snapshot build of Qt 6.4.0 returns this from `QSysInfo::prettyProductName`
[   0.689189] Frontend <Info> yuzu\main.cpp:GMainWindow:265: Host OS: Windows 10 Version 21H1

If Qt 6.3.1 shows up and is available to Open Source users I'll give it a look as well.

**Review by bunnei — APPROVED** [review]

**Comment by bunnei** [comment]
Thanks Docteh!

**Pull request merged**